### PR TITLE
Java 11 and extract Process name

### PR DESCRIPTION
**Pull request by ohadbitt**
Send process name and process user as default application name and user name on headers

### CODE REVIEW

**Comment by ohadbitt** [comment]
This includes an alternative way to extract real process name for windows but should remove next commit
 
Once removed  process name is  extracted from executed file name

**Comment by AsafMah** [comment]
The process name will always be "java.exe", so we need a method to get the main "package"/class/something.

We might want to upgrade to java 11 anyway in our next breaking change.

**Comment by ohadbitt** [comment]
Decided on making this the main module name

File: data/src/main/java/com/microsoft/azure/kusto/data/auth/ConnectionStringBuilder.java [hunk 1]
```diff
@@ -58,6 +53,16 @@ private ConnectionStringBuilder(String clusterUrl) {
...
         this.clientVersionForTracing = null;
         this.applicationNameForTracing = null;
+        this.initProcessNameForTracing();
+    }
+
+    private void initProcessNameForTracing() {
+        this.processNameForTracing = System.getProperty("sun.java.command");
+        this.userNameForTracing = System.getProperty("user.name");
+        if (processNameForTracing != null) {
+            String[] p = processNameForTracing.split(" ")[0].split("\\.");
```
Comment (AsafMah) [inline data/src/main/java/com/microsoft/azure/kusto/data/auth/ConnectionStringBuilder.java:63]: Comment explaining what's going on here.
The '\\.' is suspicious, it looks windows specific, and path specific?

Plus, make the code more defensive.
You are doing a lot of checks that can throw, and we don't want to not to be able to create a KSCB if the format of the property isn't correct, or it doesn't exist.

**Review by AsafMah — CHANGES_REQUESTED** [review]

File: data/src/main/java/com/microsoft/azure/kusto/data/auth/ConnectionStringBuilder.java (hunk 1 above)
Comment (ohadbitt) [reply data/src/main/java/com/microsoft/azure/kusto/data/auth/ConnectionStringBuilder.java:63]: What throws here

Comment (ohadbitt) [reply data/src/main/java/com/microsoft/azure/kusto/data/auth/ConnectionStringBuilder.java]: split always returns at least one element

File: data/src/main/java/com/microsoft/azure/kusto/data/UriUtils.java [hunk 2]
```diff
@@ -67,4 +67,22 @@ public static String[] getSasAndEndpointFromResourceURL(String url) throws URISy
...
+            if (processNameForTracing.equalsIgnoreCase("jar") && splitCommand.length > 1) {
+                processNameForTracing = splitCommand[splitCommand.length - 2];
+            }
+            int lastDirSeparator = processNameForTracing.lastIndexOf(File.separatorChar);
+            if (lastDirSeparator > 0) {
+                processNameForTracing = processNameForTracing.substring(lastDirSeparator + 1);
+            }
+        }
+
+        return processNameForTracing;
```
Comment (AsafMah) [inline data/src/main/java/com/microsoft/azure/kusto/data/UriUtils.java:88]:
```suggestion
        if (cmdLine == null) {
            return null;
        }

        String[] splitCommand = cmdLine.split(" ")[0].split("\\.");
        String processNameForTracing = splitCommand[splitCommand.length - 1];
        if (processNameForTracing.equalsIgnoreCase("jar") && splitCommand.length > 1) {
            processNameForTracing = splitCommand[splitCommand.length - 2];
        }
        int lastDirSeparator = processNameForTracing.lastIndexOf(File.separatorChar);
        if (lastDirSeparator > 0) {
            processNameForTracing = processNameForTracing.substring(lastDirSeparator + 1);
        }

        return processNameForTracing;
```

I think you can reduce the nesting here and make it more readable

**Review by AsafMah — APPROVED** [review]

**Comment by thewisenerd** [comment]
apologies for I am uninitiated, but is there a valid reason for breaking 1.8 compat with this change?

**Pull request merged**